nixos/modules/services/web-servers/hydron.nix

```diff
@@ -47,16 +77,40 @@ in with lib; {
   };
 
   config = mkIf cfg.enable {
+    security.sudo.enable = cfg.enable == true;
+    services.postgresql.enable = cfg.enable == true;
+
+    services.hydron.passwordFile = mkDefault (toString (pkgs.writeTextFile {
+      name = "hydron-password-file";
+      text = cfg.password;
+    }));
+
+    services.hydron.postgresArgsFile = mkDefault (toString (pkgs.writeTextFile {
+      name = "hydron-postgres-args";
+      text = cfg.postgresArgs;
+    }));
+
     systemd.services.hydron = {
       description = "hydron";
-      after = [ "network.target" ];
+      after = [ "network.target" "postgresql.service" ];
       wantedBy = [ "multi-user.target" ];
 
       preStart = ''
-        # Ensure folder exists and permissions are correct
-        mkdir -p ${escapeShellArg cfg.dataDir}/images
+        # Ensure folder exists or create it and permissions are correct
+        mkdir -p ${escapeShellArg cfg.dataDir}/{.hydron,images}
+        cp ${pkgs.hydron}/share/hydron/db_conf.json ${escapeShellArg cfg.dataDir}/.hydron
+        sed -i "s/user=hydron password=hydron dbname=hydron sslmode=disable/$(cat ${escapeShellArg cfg.postgresArgsFile})/g" \
+          ${escapeShellArg cfg.dataDir}/.hydron/db_conf.json
```

Comment:
Since this `db_conf.json` file gets copied every time, I think it's better to build the file with Nix instead and have it be a symlink to the Nix-built version. Then the user won't think of modifying it. Something like

```
ln -s ${pkgs.runCommand "db_conf.json" {} "cp && sed"} ${cfg.dataDir}/.hydron/db_conf.json
```